src/repository/users_session.rs

```diff
@@ -1,7 +1,52 @@
 use super::Repository;
-use async_session::SessionStore;
+use anyhow::Context;
+use async_session::{Session, SessionStore};
+
+use super::users::User;
 
 impl Repository {
+    pub async fn create_session(&self, user: User) -> anyhow::Result<String> {
+        let mut session = Session::new();
+        session
+            .insert("user_id", user.id)
+            .with_context(|| "Failed to insert user_id to session")?;
+        session
+            .insert("display_id", user.display_id)
+            .with_context(|| "Failed to insert display_id to session")?;
+        let result = self
+            .session_store
+            .store_session(session)
+            .await
+            .with_context(|| "Failed to store session to database")
+            .with_context(|| "Failed to create session")?;
+        match result {
+            Some(session_id) => Ok(session_id),
+            None => anyhow::bail!("unexpected error while creating session"),
+        }
+    }
+
+    pub async fn delete_session(&self, session_id: &str) -> anyhow::Result<Option<()>> {
+        let Some(session) = self
+            .session_store
+            .load_session(session_id.to_string())

```

Comment:
ここだけto_stringになってるの気になる
```suggestion
            .load_session(session_id.to_owned())

```

Reply:
全部to_string でもいい？

Reply:
全て合わせるんだったらどっちでもいいよ